cloud_functions/climateiq_csv_to_geotiff/main.py

```diff
@@ -0,0 +1,106 @@
+import os
+import pandas as pd
+from google.cloud import storage
+from io import StringIO
+import geopandas as gpd
+import rasterio
+from rasterio.transform import from_origin
+import numpy as np
+
+def convert_csv_to_geotiff(request):
+    # Get environment variables
+    bucket_name = os.getenv('BUCKET_NAME')
+    prefix = os.getenv('PREFIX')
+
+    # Initialize the storage client
+    client = storage.Client()
+    bucket = client.bucket(bucket_name)
+
+    # WKT for EPSG:32618 (NYC flood data is provided in UTM zone 18 CRS)
+    wkt_epsg_32618 = (
```

Comment:
@Rajanjain98 we have different datasets, some in UTM and some in WGS84. 
To make it more generalizable, I recommend to check first (if statement) if CRS is not epsg 32618 then do the conversion.